llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -123,6 +123,53 @@ struct HardwareLoopInfo {
   LLVM_ABI bool canAnalyze(LoopInfo &LI);
 };
 
+/// Information for memory intrinsic cost model.
+class MemIntrinsicCostAttributes {
```

Comment:
Since the initial API you're updating has a subset of these fields, please subset for now.  Specifically, only the equivalents of "unsigned Opcode, Type *Src, Align Alignment, unsigned AddressSpace" should be in this commit.